gcc/rust/ast/rust-ast.h

```diff
@@ -28,11 +28,30 @@
 
 namespace Rust {
 // TODO: remove typedefs and make actual types for these
-typedef std::string Identifier;
 typedef int TupleIndex;
 struct Session;
 struct MacroExpander;
 
+class Identifier
+{
+public:
+  Identifier (std::string ident = "")
+    : ident (ident), node_id (Analysis::Mappings::get ()->get_next_node_id ())
+  {}
+
```

Comment:
I don't think we need a node_id for an Identifier as we do resolution to the NodeId of the parent item the identifier belongs to.

I think remove the node_id, or add a copy constructor and copy assignment constructor. We also wish to have location info but feel free to do this as a separate PR see: https://github.com/Rust-GCC/gccrs/issues/2333

Reply:
> I don't think we need a node_id for an Identifier as we do resolution to the NodeId of the parent item the identifier belongs to.
> 
> I think remove the node_id, or add a copy constructor and copy assignment constructor. We also wish to have location info but feel free to do this as a separate PR see: #2333

But I do need a node Id for procedural macro resolution. When I have such derive attribute:
```
#[derive(Toto, serde::Deserialize)]
```
`serde::Deserialize` is a `SimplePath` but `Toto` is an identifier.

Reply:
Ah ok then go for it. But you will need to add the copy constructors otherwise you will get new node_id's

Reply:
> Ah ok then go for it. But you will need to add the copy constructors otherwise you will get new node_id's

Good point, will do.

Reply:
I am not completely versed in the derive macro and attribute syntax but worth double checking if rustc uses an Identifier structure there for Toto or if its using a PathExpression instead if it uses a PathExpression you might get some of the resolution steps for free